analysis/variables_cohorts.py

```diff
@@ -0,0 +1,584 @@
+        ),
+
+      ## Asthma
+        cov_bin_history_asthma_snomed= (
+            last_matching_event_clinical_snomed_before(
+            asthma_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+      ## Pulmonary Fibrosis
+        cov_bin_history_pulmonary_fibrosis_snomed= (
+            last_matching_event_clinical_snomed_before(
+            pulmonary_fibrosis_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+        ## Ischaemic stroke
+        cov_bin_stroke_isch=(
+            (last_matching_event_clinical_snomed_before(
+                stroke_isch_snomed_clinical, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_apc_before(
+                stroke_isch_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                stroke_isch_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+        ## All stroke (included the all-stroke codelists)
+        cov_bin_all_stroke=(
+            (last_matching_event_clinical_ctv3_before(
+                stroke_codes, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_clinical_snomed_before(
+                stroke_isch_snomed_clinical + stroke_sah_hs_snomed_clinical, index_date
+            ).exists_for_patient()) |           
+            (last_matching_event_apc_before(
+                stroke_isch_icd10 + stroke_sah_hs_icd10 + stroke_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                stroke_isch_icd10 + stroke_sah_hs_icd10 + stroke_icd10, index_date
+            ).exists_for_patient())
+        ),
+# Others
+    ## History of Covid-19 Combined
+
+        tmp_sub_bin_priorcovid19_confirmed_sgss=tmp_sub_bin_priorcovid19_confirmed_sgss,
+        tmp_sub_bin_priorcovid19_confirmed_snomed=tmp_sub_bin_priorcovid19_confirmed_snomed,
+        tmp_sub_bin_priorcovid19_confirmed_apc=tmp_sub_bin_priorcovid19_confirmed_apc,
+        tmp_sub_bin_priorcovid19_confirmed_opa=tmp_sub_bin_priorcovid19_confirmed_opa,
+        sub_bin_covid19_confirmed_history=(
+            tmp_sub_bin_priorcovid19_confirmed_sgss |
+            tmp_sub_bin_priorcovid19_confirmed_snomed |
+            tmp_sub_bin_priorcovid19_confirmed_apc |
+            tmp_sub_bin_priorcovid19_confirmed_opa 
+        ),
+
+    ## Covid_19 severity
+
+        sub_date_covid19_hospital = minimum_of(tmp_sub_date_severecovid19_apc, tmp_sub_date_severecovid19_opa),
+
+    # Inclusion/exclusion variables ----------------------------------------------------------------------------------------------------
+
+
+    ## Registered for a minimum of 6 months prior to the study start date # line 98: https://github.com/opensafely/comparative-booster-spring2023/blob/main/analysis/dataset_definition.py 
+
+        has_follow_up_previous_6months = (practice_registrations.for_patient_on(
```

Comment:
Do we need to check registration on index_date as well to confirm that they have the 6 months? I think this code tells you they were registered on the date six months before their index but not whether they remained registered.

Reply:
Done:
https://github.com/opensafely/post-covid-respiratory/blob/ac685d23be5d0bafe24811cf9a952878dfae5f12/analysis/variables_cohorts.py#L520-L522